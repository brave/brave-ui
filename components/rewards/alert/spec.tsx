/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Alert, { Props } from './index'

describe('Alert tests', () => {
  const baseComponent = (props: Props) => <Alert id='alert' {...props} />

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent({type: 'success'})
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent({type: 'success'}))
      const assertion = wrapper.find('#alert').length
      expect(assertion).toBe(1)
    })
  })
})
