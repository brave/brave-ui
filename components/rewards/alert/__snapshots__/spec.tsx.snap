// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert tests basic tests matches the snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background-color: #ffffff;
  box-shadow: 0 1px 0 0 #dfdfe8;
  padding: 30px 38px 33px 19px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 40px;
  height: 40px;
  -webkit-flex-basis: 40px;
  -ms-flex-preferred-size: 40px;
  flex-basis: 40px;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  padding-left: 11px;
  font-family: Poppins;
  font-size: 14px;
  line-height: 1.57;
  color: #838391;
}

.c2 b {
  font-weight: 600;
  color: #4b4c5c;
}

<div
  className="c0"
  id="alert"
>
  <img
    className="c1"
    src="test-file-stub"
  />
  <div
    className="c2"
  />
</div>
`;
